### ISSUE TITLE
Split icingaweb2 from icinga2 core

### ISSUE THREAD

**Issue by J0WI** [jjethwa/icinga2#173]
This container feels more like a VM. How about a separate icingaweb2 container that shares a database with a icinga2 instance? See e.g. https://github.com/lazyfrosch/docker-icingaweb2

**Comment by jjethwa** [comment]
Hi @J0WI 

The original idea was to keep the services together to make it easier for the average set of users.    Some work can be done to split the services needed for icingaweb2 as was done for MySQL as long as the existing (default)_ set up is supported.  Is this something you would submitting PRs for?

**Comment by J0WI** [comment]
Docker is about separation of concerns, so my idea was to use this only for icinga2, https://github.com/lazyfrosch/docker-icingaweb2 and a database container.
Users can still use a `docker-compose.yml` for a demo. Or a separate Dockefile for a demo and one without supervisord for production?

**Comment by jjethwa** [comment]
Hi @J0WI 

Well there are different schools of thought, but we can try to support both 😄   The existing setup already allows for an external database (see README).  A separate container would be needed for Apache + icingaweb2, etc.  I would still recommend using supervisord so it can act as a watchdog and shutdown processes properly.

**Comment by bebehei** [comment]
Yes, this is contradicting to the docker philosophy. I'm totally on your side.

But, in the meantime, there started a few usable docker images available. One of them is @bodsch's `bodsch/docker-icinga2`. I never used it before, but he's applying the radical docker philosophy to the icinga services and splits them in different containers.

There are also a few others out, which do this, too. It's just that @jjethwa had been the first submitter of an icinga2 docker image on the hub, so this image got the most reputation.

But back to my point: What would be the easiest way to achieve this separation? For me, it would be easier to switch images and use another image, and adjust all things I need there.

**Comment by jjethwa** [comment]
Work was started on this before docker compose or similar tooling was available, so that was why it was more of a full feature set of services offering 😛   It wouldn't be much work to add functionality of disabling Apache, then all that is left is another image that provides that functionality.
